src/org/apache/xerces/parsers/AbstractDOMParser.java

```diff
@@ -933,6 +940,10 @@ public void startElement (QName element, XMLAttributes attributes, Augmentations
                 return;
             }
             Element el = createElementNode (element);
+            // Extract location info if feature is enabled
+            if (fIncludeLocationInfo) {
+                el.setUserData( Constants.LOCATION_INFO, String.valueOf( fLocator.getLineNumber() ), null ); // Save location String into node
```

Comment:
1. These values should be Integers, not strings.
2. This might be an abuse of user data. That's meant for the application, not the parser.

Reply:
Good point. I will change them to Integers or DOM locators per your other comment. 

I would be happy to consider a different mechanism for carrying this line number data. 

This is a handy one, but I could devote an actual member on dom Elements to this. That would require users to downcast to a Xerces specific XMLEnhancedElement (say) interface to access it.  Is that preferable?

But even if we keep it on the user data, a good point also is the word "location" isn't sufficiently unique if this user data structure is being used for an internal mechanism. It should be a more unique string along the lines of a xerces internal URN.

Reply:
Changed now to using a member on NodeImpl, not user data.